scripts/package.lua

```diff
@@ -88,79 +129,101 @@
 	os.rmdir(pkgName)
 
 	print("Cloning source code")
-	z = os.executef("git clone .. %s", pkgName)
+	local z = execQuiet("git clone .. %s -b %s --recurse-submodules --depth 1 --shallow-submodules", pkgName, branch)
 	if not z then
 		error("clone failed", 0)
 	end
 
 	os.chdir(pkgName)
 
-	z = os.executef("git checkout %s", branch)
-	if not z then
-		error("unable to checkout branch " .. branch, 0)
-	end
+--
+-- Bootstrap Premake in the newly cloned repository
+--
 
-	z = os.executef("git submodule update --init")
+	print("Bootstraping Premake...")
+	if compilerIsVS then
+		z = os.execute("Bootstrap.bat " .. compiler)
+	else
+		z = os.execute("make -j -f Bootstrap.mak " .. os.host())
+	end
 	if not z then
-		error("unable to clone submodules", 0)
+		error("Failed to Bootstrap Premake", 0)
 	end
+	local premakeBin = path.translate("bin/release/Premake5")
 
 
 --
 -- Make absolutely sure the embedded scripts have been updated
 --
 
 	print("Updating embedded scripts...")
-	if kind == "source" then
-		z = execQuiet("premake5 embed")
-	else
-		z = execQuiet("premake5 --bytecode embed")
-	end
+
+	local z = execQuiet("%s embed %s", premakeBin, iif(kind == "source", "", "--bytecode"))
 	if not z then
 		error("failed to update the embedded scripts", 0)
 	end
 
 
 --
--- Clear out files I don't want included in any packages.
+-- Generate a source package.
 --
 
-	print("Cleaning up the source tree...")
-	os.rmdir("packages")
-	os.rmdir(".git")
+if kind == "source" then
 
-	local removelist = { ".DS_Store", ".git", ".gitignore", ".gitmodules", ".travis.yml", ".editorconfig", "appveyor.yml", "Bootstrap.mak" }
-	for _, removeitem in ipairs(removelist) do
-		local founditems = os.matchfiles("**" .. removeitem)
-		for _, item in ipairs(founditems) do
-			os.remove(item)
+	local function	genProjects(parameters)
+		if not execQuiet("%s %s", premakeBin, parameters) then
+			error("failed to generate project for "..parameters, 0)
 		end
 	end
 
---
--- Generate a source package.
---
-
-if kind == "source" then
+	os.rmdir("build")
 
 	print("Generating project files...")
-	execQuiet("premake5 /to=build/vs2005 vs2005")
-	execQuiet("premake5 /to=build/vs2008 vs2008")
-	execQuiet("premake5 /to=build/vs2010 vs2010")
-	execQuiet("premake5 /to=build/vs2012 vs2012")
-	execQuiet("premake5 /to=build/vs2013 vs2013")
-	execQuiet("premake5 /to=build/vs2015 vs2015")
-	execQuiet("premake5 /to=build/vs2017 vs2017")
-	execQuiet("premake5 /to=build/vs2019 vs2019")
-	execQuiet("premake5 /to=build/gmake.windows /os=windows gmake")
-	execQuiet("premake5 /to=build/gmake.unix /os=linux gmake")
-	execQuiet("premake5 /to=build/gmake.macosx /os=macosx gmake")
-	execQuiet("premake5 /to=build/gmake.bsd /os=bsd gmake")
+
+	local ignoreActions = {
+		"clean",
+		"embed",
+		"gmake2",
+		"package",
+		"test",
+
+		"codelite",
+		"xcode4",
```

Comment:
After this release, we should consider bundling these two project types too. I would imagine that `codelite` would be similar to `gmake`.